lighthouse-core/lib/navigation-error.js

```diff
@@ -0,0 +1,145 @@
+    });
+  }
+}
+
+/**
+ * Returns an error if we ended up on the `chrome-error` page and all other requests failed.
+ * @param {LH.Artifacts.NetworkRequest|undefined} mainRecord
+ * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+ * @return {LH.LighthouseError|undefined}
+ */
+function getInterstitialError(mainRecord, networkRecords) {
+  // If we never requested a document, there's no interstitial error, let other cases handle it.
+  if (!mainRecord) return undefined;
+
+  const interstitialRequest = networkRecords.find(record =>
+    record.documentURL.startsWith('chrome-error://')
+  );
+  // If the page didn't end up on a chrome interstitial, there's no error here.
+  if (!interstitialRequest) return undefined;
+
+  // If the main document didn't fail, we didn't end up on an interstitial.
+  // FIXME: This doesn't handle client-side redirects.
+  // None of our error-handling deals with this case either because passContext.url doesn't handle non-network redirects.
+  if (!mainRecord.failed) return undefined;
+
+  // If a request failed with the `net::ERR_CERT_*` collection of errors, then it's a security issue.
+  if (mainRecord.localizedFailDescription.startsWith('net::ERR_CERT')) {
+    return new LHError(LHError.errors.INSECURE_DOCUMENT_REQUEST, {
+      securityMessages: mainRecord.localizedFailDescription,
+    });
+  }
+
+  // If we made it this far, it's a generic Chrome interstitial error.
+  return new LHError(LHError.errors.CHROME_INTERSTITIAL_ERROR);
+}
+
+/**
+ * Returns an error if we try to load a non-HTML page.
+ * Expects a network request with all redirects resolved, otherwise the MIME type may be incorrect.
+ * @param {LH.Artifacts.NetworkRequest|undefined} finalRecord
+ * @return {LH.LighthouseError|undefined}
+ */
+function getNonHtmlError(finalRecord) {
+  // MIME types are case-insenstive but Chrome normalizes MIME types to be lowercase.
+  const HTML_MIME_TYPE = 'text/html';
+
+  // If we never requested a document, there's no doctype error, let other cases handle it.
+  if (!finalRecord) return undefined;
+
+  // mimeType is determined by the browser, we assume Chrome is determining mimeType correctly,
+  // independently of 'Content-Type' response headers, and always sending mimeType if well-formed.
+  if (HTML_MIME_TYPE !== finalRecord.mimeType) {
+    return new LHError(LHError.errors.NOT_HTML, {mimeType: finalRecord.mimeType});
+  }
+
+  return undefined;
+}
+
+/**
+ * Returns an error if the page load should be considered failed, e.g. from a
+ * main document request failure, a security issue, etc.
+ * @param {LH.LighthouseError|undefined} navigationError
+ * @param {{url: string, loadFailureMode: LH.Gatherer.PassContext['passConfig']['loadFailureMode'], networkRecords: Array<LH.Artifacts.NetworkRequest>}} context
+ * @return {LH.LighthouseError|undefined}
+ */
+function getPageLoadError(navigationError, context) {
+  const {url, loadFailureMode, networkRecords} = context;
```

Comment:
Should `loadFailureMode` available in the FR config? Maybe in the navigation config?

Reply:
yep it will be, it's part of `SharedPassNavigationJson` in the config